cmd/demo/demo_test.go

```diff
@@ -46,40 +52,102 @@ func nodeCmd(name string) (*Cmd, error) {
+	// Start Alice.
 	alice, err := nodeCmd("alice")
 	require.NoError(t, err)
 	require.NoError(t, alice.Start())
 	defer alice.Process.Kill()
-	time.Sleep(blockTime * time.Duration(2)) // wait 2 blocks for contract deployment
+	time.Sleep(blockTime * 2) // Wait 2 blocks for contract deployment.
 
-	// Start Bob
+	// Start Bob.
 	bob, err := nodeCmd("bob")
 	require.NoError(t, err)
 	require.NoError(t, bob.Start())
 	defer bob.Process.Kill()
-	time.Sleep(5 * time.Second) // give bob some time to initialize
+	time.Sleep(5 * time.Second) // Give Bob some time to initialize.
+
+	// Set the initial on-chain balances from Alice and Bob.
+	initBals, err := getOnChainBals()
+	require.NoError(t, err)
+	t.Logf("Initial on-chain balances: Alice = %f, Bob = %f", initBals[0], initBals[1])
 
-	// Alice opens channel with Bob
+	// Alice opens channel with Bob.
 	require.NoError(t, alice.sendCommand("open bob 1000 1000\n"), "proposing channel")
+	time.Sleep(3 * time.Second) // Ensure that Bob really received the proposal.
 	require.NoError(t, bob.sendCommand("y\n"), "accepting channel proposal")
 	t.Log("Opening channel…")
-	time.Sleep(blockTime) // wait 1 block for funding transactions to be confirmed
+	time.Sleep(blockTime) // Wait 1 block for funding transactions to be confirmed.
 
-	// Alice sends to Bob and Bob to Alice
-	for i := 0; i < 25; i++ {
+	// Alice sends to Bob and Bob to Alice.
+	for i := 0; i < sendIterations; i++ {
 		t.Log("Sending payment… (alice->bob)")
 		require.NoError(t, alice.sendCommand("send bob 1\n"))
 		time.Sleep(100 * time.Millisecond)
 		t.Log("Sending payment… (bob->alice)")
 		require.NoError(t, bob.sendCommand("send alice 2\n"))
 		time.Sleep(100 * time.Millisecond)
 	}
+
+	t.Log("Closing channel…")
+	require.NoError(t, alice.sendCommand("close bob\n"))
+	// Wait 2 blocks for the settle and withdrawal transactions plus some additional seconds.
+	time.Sleep(2*blockTime + 5*time.Second)
+
+	// Set the final balances from Alice and Bob after the settlement.
+	finalBals, err := getOnChainBals()
+	require.NoError(t, err)
+
+	t.Logf("Final on-chain balances: Alice = %f, Bob = %f", finalBals[0], finalBals[1])
+
+	var diffBals [2]float64
+
+	// Calculate the differences between the initial and final balances.
+	diffBals[0], _ = finalBals[0].Sub(finalBals[0], initBals[0]).Float64()
+	diffBals[1], _ = finalBals[1].Sub(finalBals[1], initBals[1]).Float64()
+
+	// Check the on-chain balance differences while allowing a higher deviation for Alice.
+	require.Less(t, diffBals[0], sendIterations+0.1)
+	require.Greater(t, diffBals[0], sendIterations-0.1)
+	require.Less(t, diffBals[1], -sendIterations+0.01)
+	require.Greater(t, diffBals[1], -sendIterations-0.01)
+
 	t.Log("Done")
 }
 
 func (cmd *Cmd) sendCommand(str string) error {
 	_, err := io.WriteString(cmd.stdin, str)
 	return err
 }
+
+func getOnChainBals() ([2]*big.Float, error) {
+	var bals [2]*big.Float
+
+	ctx, cancel := context.WithTimeout(context.Background(), time.Second*30)
```

Comment:
Did you try it with lower timeout? 30s seems a bit excessive for just two queries. 3s should work too.  
You can also keep it and use `DialContext` below instead of `Dial` then it makes more sense AFAIK.